File: pekko-sample-sharding-scala/killrweather/src/main/scala/sample/killrweather/Guardian.scala
package sample.killrweather

import org.apache.pekko.actor.typed.Behavior
import org.apache.pekko.actor.typed.scaladsl.Behaviors

/**
 * Root actor bootstrapping the application
 */
object Guardian {

  def apply(httpPort: Int): Behavior[Nothing] = Behaviors.setup[Nothing] { context =>
    WeatherStation.initSharding(context.system)

    val routes = new WeatherRoutes(context.system)
    WeatherHttpServer.start(routes.weather, httpPort, context.system)

    Behaviors.empty
  }

}

File: pekko-sample-sharding-java/killrweather/src/main/java/sample/killrweather/Guardian.java
package sample.killrweather;

import org.apache.pekko.actor.typed.Behavior;
import org.apache.pekko.actor.typed.javadsl.Behaviors;

/**
 * Root actor bootstrapping the application
 */
final class Guardian {

  public static Behavior<Void> create(int httpPort) {
    return Behaviors.setup(context -> {
      WeatherStation.initSharding(context.getSystem());

      WeatherRoutes routes = new WeatherRoutes(context.getSystem());
      WeatherHttpServer.start(routes.weather(), httpPort, context.getSystem());

      return Behaviors.empty();
    });
  }
}
